lib/rubocop/cop/rspec/empty_line_after_final_let.rb

```diff
@@ -29,24 +31,32 @@ def on_block(node)
           return if latest_let.nil?
           return if latest_let.equal?(node.body.children.last)
 
-          no_new_line_after(latest_let) do
-            add_offense(latest_let, location: :expression)
+          no_new_line_after(latest_let) do |location|
+            add_offense(latest_let, location: location)
           end
         end
 
         def autocorrect(node)
-          loc = last_node_loc(node)
-          ->(corrector) { corrector.insert_after(loc.end, "\n") }
+          lambda do |corrector|
+            no_new_line_after(node) do |location|
+              corrector.insert_after(location.end, "\n")
+            end
+          end
         end
 
         private
 
         def no_new_line_after(node)
           loc = last_node_loc(node)
+          line = loc.line
+          has_comment = false
+          while comment_line?(processed_source[line])
+            line += 1
+            has_comment = true
+          end
```

Comment:
I wonder if this cannot somehow be simplified. We want two things here, a boolean `has_comment` that is true if `loc.line` is a comment; and `line` which I guess is the last line of a multi-line comment. Could we have some code like this:
```rb
has_comment = comment_line?(processed_source[line])
last_commented_line = line + commented_lines_from(line) - 1
```
Also, bonus points if we can avoid `.offending_loc` taking a flag argument :-)